depends/patches/systemtap/remove_SDT_ASM_SECTION_AUTOGROUP_SUPPORT_check.patch

```diff
@@ -0,0 +1,31 @@
+commit b92d4c121486f3c6e8a2cea537c53eb09894479a
+Author: 0xb10c <0xb10c@gmail.com>
+Date:   Tue Dec 7 11:02:07 2021 +0100
+
+    Remove _SDT_ASM_SECTION_AUTOGROUP_SUPPORT check
+    
+    We assume that the assembler supports "?" in .pushsection directives.
```

Comment:
After looking at this again, I think assuming the assembler supports `?` is ok, assuming we make this Linux only (see other review comments).

`pushsection` is an [ELF specific directive](https://sourceware.org/binutils/docs/as/PushSection.html). If you try to compile the assembly used in the configure check which would define `_SDT_ASM_SECTION_AUTOGROUP_SUPPORT`, with `mingw32-gcc`, it won't compile:
```c
int main() {

asm(".section .note.foo,\"?\",\"note\"\n"
     ".byte 1, 2, 3\n"
     ".previous\n"
     ".section .text,\"axG\",\"progbits\",foogroup,comdat\n"
     ".byte 1\n"
     ".pushsection .note.foo,\"?\",\"note\"\n"
     ".byte 4, 5, 6\n"
     ".popsection\n"
     ".byte 2\n");
return 0;
}
```

```bash
x86_64-w64-mingw32-gcc push.c 
/tmp/ccDQdocw.s: Assembler messages:
/tmp/ccDQdocw.s:18: Warning: unknown section attribute '?'
/tmp/ccDQdocw.s:18: Error: junk at end of line, first unrecognized character is `,'
/tmp/ccDQdocw.s:20: Error: unknown pseudo-op: `.previous'
/tmp/ccDQdocw.s:21: Warning: unknown section attribute 'G'
/tmp/ccDQdocw.s:21: Warning: Ignoring changed section attributes for .text
/tmp/ccDQdocw.s:21: Error: junk at end of line, first unrecognized character is `,'
/tmp/ccDQdocw.s:23: Error: unknown pseudo-op: `.pushsection'
/tmp/ccDQdocw.s:25: Error: unknown pseudo-op: `.popsection'
```
I think making the systemtap package Linux only for now is the way to go.